### ISSUE TITLE
What is needed the most?

### ISSUE THREAD

**Issue by Danimal696** [OpenDungeons/OpenDungeons#1228]
Im going to get a lot more of free time soon, so i was wondering, what is OD in most urgent need? 
We are pacing slower these days, but its normal due to time restrictions and good weather, but @OpenDungeons/developers what do you think its the aspect most deserving of attention to progress?. Feel free to add yours:
- More game models
- More rooms and game elements (like mana fountain, Temple, forge...)
- Adding music and sounds
- More polish (like better model scaling, new or upgraded tilesets/textures)
- More publicity/exposure (i think we are confident in the current build)
- More levels or a Campaign
- Some planification of things to come

For the time being im animating the fairy, but i would like your input.

**Comment by hwoarangmy** [comment]
IMHO, what is needed the most:
- A campaign to be able to release a 1.0 version
- A way to be able to attack an enemy who is turtling (as discussed on #1184) as ATM, such an enemy cannot be attacked at all.

For my part, the lack of activity lately (except @Loobinex who has btw interesting ideas about game mecanics) has greatly reduced my motivation. I will probably continue to make some bug fixes but I don't think I will implement major features for some time. I'm not complaining, everybody is free to spend the freetime he wants (and has) where he wants. But IMHO, the project do not have enough active contributors ATM to continue.

**Comment by akien-mga** [comment]
What @hwoarangmy points out is sad but true, over the last few months @Bertram25 and I have been mostly missing (my free time is pretty much split between contributing to @godotengine and to Mageia, not much time for other projects sadly) and @hwoarangmy has done most of the 0.7.0 development alone. It's understandably not motivating to put much work into the project.

I wouldn't consider the project dead though, but the development pace can't be what it was in 2014-2015 when we revived the project with a new team. Partly due to our lack of communication, we dearly miss actual people that play the game, and it's hard to sustain the motivation to develop a game that nobody plays.

We should of course still publish 0.7.0, make some noise about it and tell the world that we welcome new contributors, and we'll see what happens. @Loobinex has joined us recently with some great ideas, @Danimal696 promises to have some more time in the near future, and the rest of us developers are still more or less around. If we get a small players community that open bug reports, and a couple contributors more, I'm pretty sure the development would restart full steam ahead :)

**Comment by Loobinex** [comment]
I don't think a campaign should be high at the todo list at this point. Mechanics come before content.
When all the mechanics are finished you can make a campaign to make the most those mechanics, and allow other users to add more content with the map editor.

To be honest, I don't think the current game is that fun to play, and I don't think I would be the only one that would rather play one of the DK games or even some of the clones. It currently all feels a bit clunky and there's nothing really that stands out, so you don't get much of an audience, and will also have more difficulty in finding new people to help out with development/playtesting/mapmaking.

I think focus should be on polishing the core mechanics already there - to make them feel right to a player. Little interface (sound, controls, timings, animations etc) tweaks that make simply building the dungeon and managing the creatures already be enjoyable.

**Comment by hwoarangmy** [comment]
> To be honest, I don't think the current game is that fun to play

Sadly, this sounds pretty true. I guess part of it comes from the fact the levels are too scattered with water/lava as you already noticed.
IMHO, we are also lacking real gameplay. Almost every creatures comes from the same rooms and the strategy part is also missing. We have the same problem as the original DK: the only strategy is to rush for top level creatures and try to get as many as possible.

**Comment by Loobinex** [comment]
Yeah, I believe I already made a post about that around here somewhere, but currently 'getting stronger creatures through dungeon keeping' is not what the game is about, it just happens.
The portal limit(if you get 15 creatures, no matter what you do, you want 15 strong creatures), basic attraction requirements and tier system make it almost inevitable.

**Comment by Danimal696** [comment]
So, what can be done? tying high tier creatures to expensive and late game rooms is a given.
Perhaps increasing the HP pool of all of them so they dont fall like flies and combats last longer, something to make you care about them? or maybe the opposite, make them even more expendable so you only have to care about progressing the map.

It has to be decided which way to go, if we go for fast games, the last i said is true, everything is cannon fodder for us; or we migth want to focus on the joy of dungeon building and a slower pace. Which involves giving the player more things to do on his dungeons, be it dealing with creatures or managing rooms prisioners...

Two good examples are Dungeons 2  for slow pace and dungeonering and War for the overworld for furious fast pace. Give a look at some of both gameplays if you can.

**Comment by Loobinex** [comment]
A lot can be done, but it's no quick fix. And yes, first you need to decide what kind of game you want (remember #1147), and then you can determine which steps are needed to get there. Unfortunately, there's not really a template out there to use when making a 'dungeon game'. Both the original DK games were rushed, Dungeons was there mostly to capitalize on the name and is indeed a lot slower and riddled with problems, and WftO was a DK-fangame left to be taken over by a new 'CEO/lead designer/project manager/producer' with no design/management experience or competence resulting in fundamentally flawed gameplay. There have been some fun games in the genre - I liked evil genius for example - but they are all quite different from each other.

Personally I don't think a 'dungeon game' lends itself to be a fast-paced multiplayer game. RTS games revolve around strategizing a plan of attack, micro managing your units, and quick and smart reaction to the battle plans and attacks of your rivals.
The core of dungeon games - and most people don't even realize this - is not 'being evil' or 'being underground'; It is having creatures with a mind of their own. Lack of control is key. I remember an interview I read on the back of DK1 when Molyneux was already talking up Black&White that his inspiration for DK was his feeling he had as a child closely observing ants going about their business, and both enjoying giving them sugar or burning them with a magnifying glass. And this shows in the game, creatures come almost randomly into your dungeon, and when you're lucky they do what you want and fight for you. You don't order them around, you punish or reward them. Now tell me how to unify these concepts,... Random units with a mind of their own and the fact that there's a metric ton of dirt, rocks and potential dangers between you and your rivals just don't lend themselves to the same kind of gameplay scenario's that make a game like Starcraft fun.

What I think you need is that when you start a level that the **(1)**building and designing the dungeon is fun, and that after the initial phase you'll start to get some pressure somehow. This forces you to explore the map, go do stuff there, risk your valuable creatures for a needed advantage(gold, new creatures, new tech, something to run or **(2)**improve your dungeon), until **(3)**finally you'll survive an invasion of heroes or invade a rival.
Would you agree that playing a level of dungeons should follow such a structure? If so you can with this structure in mind add features/make changes to support each individual step. If not, come up with another structure and do the same.

**Comment by Loobinex** [comment]
Adding to the above, making a multiplayer focused game is possible of course. No need to make it a RTS game. There's lots of other options.
You need to focus the competitive elements around the strengths of the genre however. Both players should compete for creatures, where having the best dungeon will attract creatures from a limited but shared pool. And being able to push out on the map is both risky and rewarding. Strategic targets need to be visible and you need to be able to contest them.
For multiplayer combat needs to be completely revamped though, as right now if you lose one battle, you lose the game.
It's also possible to make it mostly a coop game, where you have to defend against invading heroes together, again trying to balance defending, building and exploring the map/attacking.

**Comment by hwoarangmy** [comment]
Well, of course, we are not aiming for a Starcraft RTS like game play. But I think we should somehow try to find a way to make the game startegic so that it is not just a dice throw and you might win or loose depending on luck only (actually, I would even prefer to not rely on luck at all).

Concerning game length, when I joined the project, I was hesitating between OD and a settlers 2 clone called Widelands. I chose OD because Widelands games just last forever. I like the concept and I was a settlers 2 fan. They added some interesting stuff, especially about military and new buildings but the games are really too slow. I think we should aim for shorter games. IMHO, a game should last an average of 30 minutes up to 1 hour if the players prefer to upgrade instead of fighting. For my part, I will not play a multiplayer game if I know I will have to be here for more than 1 hour (that's why I never played Widelands multiplayer games).

From the start, IMHO, gold is the tool that allows us to tune player aggressivity. The more the creatures (and rooms/traps) use gold, the more the player will have to find more (that's why gem digging is a big problem ATM IMHO).

@Loobinex When I see your complaints, I think that most of the things you have noticed are about level conception and not about the game mecanics themselves.

Regarding creature fights, they are not too bad IMHO as they currently are. Not too fast and not too slow. But in the end game, it becomes pretty annoying to have to drop the creatures in front of an enemy dungeon and see all creatures go back to get their fee/eat/sleep.

**Comment by Loobinex** [comment]
I'm not complaining, I'm helping. :)

And yes, the mechanics need tweaking to get fun gameplay, but you need to be clear on the high level concept before you can determine the mechanics to support them. And seeing Danimal696s comment, I really don't think you have your concept figured out.

Knowing that you want a multiplayer focused game with strategic choices and 30-60 minutes games is a start, but allows for all kinds of gameplay. If not Starcraft, then what?(I'm not saying you should do starcraft).
I'm serious, tell me which phases(3 or 4) there are in these 30-60 minutes, and why the player is having fun in each of them.

**Comment by hwoarangmy** [comment]
> I'm not complaining, I'm helping. :)

That was not meant to be negative. I could have said "the flaws you've seen" ;-)

BTW, without many thoughts, I find the idea of the shared spawn pool interesting. The cons being that it would make you know easily what your opponents are going for (especially in 1v1). But it would allow to add some use to most basic creatures (since when there are no more high level ones, you will have no choice) and it will make you have to care more about saving them.

We could also have a fixed spawn pool (at game start, same for everybody but when a creature spawns, it is decreased). That would hide what opponents are doing but it would remove the strategy of emptying some creatures you don't want your opponent to have.

We could also have something similar to starcraft games. When you reach the minimal needs for a creature type, you could choose it in a list and it would spawn after a while. Of course, the most powerfull the creature is, the most time it will take to come. And this can work with the fixed spawn pool as well.

@OpenDungeons/developers WDYT ?

**Comment by Danimal696** [comment]
War for the overworld use a pretty fresh approach to getting cretures; for example basic rooms like lair and hatchery dont attract anything but are needed to keep creatures happy, the ones really attracting creatures are rooms like Training room = goblins; library = mages; beast den = beasts (duh!) and so on.

So the rooms will actually attracts X creatures based on their size and limited to a total creature cap. 
No randomness there, a dungeon with a 5x5 training room = 8 goblins (2 per active spot)... and so on, the beast den allows you to decide what beasts comes in (bats-things or spider-things...)
So basically you decide what you want by building their rooms and creature population runs on auto-mode from then on, there are some dependancies, like the beast master will only join you if you build the arena and have beasts... the big guys have several other dependiencies.

Its pretty liberating and helps have fast games with a tailored army, less luck involved, money grabbing is pretty hectic at the start, portals functions is to regulate creature spawn rate (more portal = better refresment of dead troops) and raise the total creature limit.

**Comment by Loobinex** [comment]
When you have a shared pool a correctly functioning creature attraction algorithm should prevent you from knowing what your opponent is going for, at least until it is too late. And players should not know the size of the pool.

I think only low and mid tier creatures should be attracted simply by owning rooms, but the attraction force should be calculated by including both the number of creatures you already have, and how much larger your room is compared to that of your rivals. (E.g. When I have a 25training and 0 goblins but my rival has 30 training and 8 goblins, the next goblin should go to me, but if we would have equal numbers he would go to him.)
Notice this would reverse the system from 'The player should get a creature now, draw one from the list of possibilities' to 'a creature becomes available to be attracted, determine to which -if any - player it goes'. This is much more in line with the concept of competitive dungeon building.

You should also incur penalties on your attractiveness score for a creature type - you killing them directly should be a major one, some others should be things like losing them in battle, not paying them, having them unhappy, owning many creatures it hates.

This all would start the battle before the actual fighting - which is what you call strategy.

**Comment by hwoarangmy** [comment]
That could indeed add some salt to the start of the game.
So, what about this logic:
- The spawn pool is fixed and unknown from players (once it is empty, only auto spawning creatures like skeletons or such can come to the dungeon - which would increase the need of similar rooms btw)
- At a fixed rate (depending on the seat number), a creature in the spawn pool is randomly chosen and the points for spawning in tested for every seat with a portal to know where the creature should spawn. It will always spawn in the seat that gets the more points.
- If no seat meets the requirements, the creature do not spawn.

The cons:
- When the spawn pool will be empty from interesting creatures, the game will become very boring
- Decreases the importance of portals

The pros:
- Make the game more strategic at dungeon building. For example, most basic creature will like a big lair. So, if you spend money building one, you are likely to get many creatures at game start when your opponents won't have allowing some rush tactic.
- Add a real competition for dungeon building

**Comment by Loobinex** [comment]
Sounds like a good start. To combat the negatives and add an additional layer what about:
- Distance from the room to the portal affects attraction rate. With multiple portals each portal gives another attraction score so you have a bigger chance to attract creatures (e.g. the portal close to training room and library attracts goblins and cultist, and the one close to workshop will attract trolls). This will also force players more to think about dungeon layouts.
- Include a way for the mapmaker to add creatures to the pool based on certain conditions, like time played, creatures defeated, or map area reached.

But also note is that you can have a very large pool and that the ratio of the creature-types inside determine how many of each type will spawn. This way you'll never run out but still compete for creatures with rivals.
I think this also suggests a creature limit per map instead of per player, which again may also be increased by level-events.

**Comment by Loobinex** [comment]
With the recently implemented changes, I'd say the single most needed feature now is a full-map screen (#1151).

**Comment by hwoarangmy** [comment]
I might have a look at this as I agree it would be a nice addition. Probably will I add a new type of minimap to not mess with the current ones.
